### PR TITLE
feat: warn when operating on destroyed Vue component, fix #1685

### LINKED ISSUE #1685: Throw errors when operating on destroyed wrapper
A bit related to #1342 

While it is obviously impossible to "destroy" something in JavaScript it could be great help to make wrappers throw error when operating on destroyed Vue instance. I suggest making `emitted` (and deprecated `emittedByOrder`) the only exceptions, cause I can see use cases, when we want to assert instance generated certain events during destroy process.

This will help a lot when introducing `enableAutoDestroy(beforeEach)` in big codebases to track such hard-to-debug errors
I'm ready to submit PR for this one if team feels this one like a good addition

### PR DESCRIPTION
**Pull request by xanf**
See #1685 for original idea
In general, I've decided to go with warning as more relaxed way instead of throwing

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [X] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [X] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [X] It's submitted to the `dev` branch.
- [X] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] All tests are passing: https://github.com/vuejs/vue-test-utils/blob/dev/.github/CONTRIBUTING.md#development-setup
- [x] New/updated tests are included

If adding a **new feature**, the PR's description includes:

- [X] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

### CODE REVIEW

File: test/specs/wrapper/destroy.spec.js [hunk 1]
```diff
@@ -61,4 +74,49 @@ describeWithShallowAndMount('destroy', mountingMethod => {
     const wrapper = mountingMethod(TestComponent)
     expect(() => wrapper.destroy()).toThrow()
   })
+
+  const StubComponent = { template: '<div><p></p></div>' }
+
+  ;[
```
Comment (xanf) [inline test/specs/wrapper/destroy.spec.js:80]: It would be great to use `jest.each` here, but that will fail in Karma :(

Comment (lmiller1990) [reply test/specs/wrapper/destroy.spec.js:80]: A lot of cool jest things fail in karma... all good.

**Review by lmiller1990 — APPROVED** [review]
nice and simple

**Comment by lmiller1990** [comment]
@xanf thanks I can release this weekend!

**Pull request merged**